app.rb

```diff
@@ -122,7 +128,12 @@ def config
 
     private
 
-    def authorization_url(ial:, aal: nil)
+    def authorization_url(session:, ial:, aal: nil)
+      if ial == 'step-up'
+        ial = '1'
+        session[:step_up_enabled] = 'true'
+        session[:step_up_aal] = aal
+      end
```

Comment:
Should we clear the session value if the user revisits the page without the `step-up` parameter?

```suggestion
      if ial == 'step-up'
        ial = '1'
        session[:step_up_enabled] = 'true'
        session[:step_up_aal] = aal
      else
        session.delete(:step_up_enabled)
        session.delete(:step_up_aal)
      end
```

Reply:
Done in e65cbd6